### DISCUSSION TITLE
Automation loop stops after 200

### DISCUSSION THREAD

**Discussion by m7kael** [Budibase/budibase#10099] [Help] [answered]
Hi,
I have an automation that queries around 500 rows, then loops through them and updates one field. The automation works but it stops after 200 loops. Is there a loop limit or am I doing something wrong? 

I have tried changing the max loop iterations to 500 but the same result. 

![Screenshot 2023-03-22 at 18 42 17](https://user-images.githubusercontent.com/112936118/226991935-36606a6b-68d4-4eb9-9fc2-0d9a111627d2.png)
![Screenshot 2023-03-22 at 18 35 25](https://user-images.githubusercontent.com/112936118/226992549-5e614761-6428-4269-9420-f741c899f1d5.png)

**Comment by ConorWebb96** [comment, marked as answer]
Hey @m7kael,

Budibase automations support "Looping" (https://docs.budibase.com/docs/looping). However, there is a hard limit of 200 iterations (loops) at one time. So 200 would be the maximum that you could loop through at one time.

I hope this helps!

**Reply by Shadowfita** [reply]
Can we change this 200 maximum when self-hosting?

**Reply by ConorWebb96** [reply]
Hey @Shadowfita,

I believe theres an .env variable `AUTOMATION_MAX_ITERATIONS ` which you can set to override the 200 automation loop limit. As you've said this will only work for self-hosted users.

I hope this helps.

**Reply by Shadowfita** [reply]
Hey @ConorWebb96,

Thank you for that. That worked a treat. Now to figure out if I can extend the automation timeout...
